src/engine/strat_engine/backstore/blockdevmgr.rs

```diff
@@ -427,6 +454,67 @@ mod tests {
         );
     }
 
+    /// Test that the `BlockDevMgr` will not add devices if a different key
+    /// is present in the keyring than was used to encrypted the existing
+    /// devices.
+    fn test_blockdevmgr_changed_key(paths: &[&Path]) {
+        fn test_with_first_key(
+            paths: &[&Path],
+            key_desc: &str,
+            _: Option<()>,
+        ) -> Result<(PoolUuid, BlockDevMgr), Box<dyn Error>> {
+            let pool_uuid = Uuid::new_v4();
+            let bdm = BlockDevMgr::initialize(
+                pool_uuid,
+                &paths[..2],
+                MDADataSize::default(),
+                Some(key_desc.to_string()),
+            )?;
```

Comment:
Please verify add with same key works here,  to get both paths covered.

Reply:
Resolved.